### PR TITLE
testing: tests were not being run on all dbs

### PR DESCRIPTION
**Pull request by marbar3778**
- added tsting for all dbs
- this should of been caught earlier
- will add this to ci as well

Signed-off-by: Marko Baricevic <marbar3778@yahoo.com>

### CODE REVIEW

**Comment by marbar3778** [comment]
prioir to the interface errors refactor I believe rocksdb was not abiding by the interface as well. This will help avoid all of this

**Comment by marbar3778** [comment]
this is blocked on getting a docker image produced. have a PR here: https://github.com/tendermint/images/pull/37

**Comment by marbar3778** [comment]
add codecov in with this PR since ci is being touched, it would be useful

File: makefile [hunk 1]
```diff
@@ -7,10 +7,27 @@ export GO111MODULE = on
...
+	@echo "--> Running go test"
+	@go test -p 1 $(PACKAGES) -tags rocksdb
+
+test-boltdb:
+	@echo "--> Running go test"
+	@go test -p 1 $(PACKAGES) -tags boltdb
+
+test-all:
+	@echo "--> Running go test"
+	@go test -p 1 $(PACKAGES) -tags cleveldb,boltdb
```
Comment (marbar3778) [inline makefile]:
```suggestion
	@go test -p 1 $(PACKAGES) -tags cleveldb,boltdb,rocksdb
```
add back rocksdb

**Comment by erikgrinaker** [comment]
What is remaining here?

**Comment by marbar3778** [comment]
just need to pass the correct path to rocksdb when running the tests

**Comment by erikgrinaker** [comment]
Do you mind if I pick this up? I'm seeing some bugs in the other backends that would be caught by the common test suite.

**Comment by marbar3778** [comment]
yea for sure!! thank you

**Comment by erikgrinaker** [comment]
Closing in favor of #64, since most changes here have been ported to `master` separately.

**Pull request closed without merging**